cpp/src/parquet/arrow/reader.h

```diff
@@ -187,6 +188,11 @@ class PARQUET_EXPORT FileReader {
       const std::vector<int>& row_group_indices, const std::vector<int>& column_indices,
       std::unique_ptr<::arrow::RecordBatchReader>* out) = 0;
 
+  virtual ::arrow::Status GetRecordBatchReader(
+      const std::vector<int>& row_group_indices, const std::vector<int>& column_indices,
+      const std::shared_ptr<std::map<int, RowRangesPtr>>& row_ranges_map,
```

Comment:
So, does `RowRanges` index means row-idx in this RowGroup or the row-idx in this file?

Reply:
row-idx in this RowGroup

Reply:
Please document this in the method contract.  The way this is being passed (as a shared pointer and values being pointer seems a little strange to me) is there a reason how this contract was arrived at (I need to take a closer look at how Range was arrived at).

I'm not sure if keeping consistency for a brand new method is worth it or if this should return arrow::Result<std::unique_ptr<RecordBatchReader>> at some point we should convert all the methods to return result.

Reply:
Its also not clear to me that we should include both RowGroupIndices and ranges within those indicies.  It seems like a cleaner API to only include one or the other (which also makes passing in ranges easier?)

Reply:
changed to File based